SYCL/Basic/code_location_e2e.cpp

```diff
@@ -0,0 +1,142 @@
+// RUN: %clangxx -fsycl -DNDEBUG %s -o %t.out
+// RUN: %CPU_RUN_PLACEHOLDER %t.out %CPU_CHECK_PLACEHOLDER
+// RUN: %GPU_RUN_PLACEHOLDER %t.out %GPU_CHECK_PLACEHOLDER
+// RUN: %ACC_RUN_PLACEHOLDER %t.out %ACC_CHECK_PLACEHOLDER
+
+// RUN: %clangxx -fsycl %s -o %t.out
+// RUN: %CPU_RUN_PLACEHOLDER %t.out %CPU_CHECK_PLACEHOLDER
+// RUN: %GPU_RUN_PLACEHOLDER %t.out %GPU_CHECK_PLACEHOLDER
+// RUN: %ACC_RUN_PLACEHOLDER %t.out %ACC_CHECK_PLACEHOLDER
+
+/*
+    clang++ -fsycl -DNDEBUG -o smyl.bin code_location_e2e.cpp   // <<--- NDEBUG
+   will suppress fileLocation
+
+    -DNDEBUG  <-- not debugging, meaning fileName should not be retrievable (for
+   data privacy reasons)
+
+    This test uses hardcoded file code location positions. If modified, those
+   may change. pass in any value to signal "report only" and not run tests.
+
+*/
+
+// clang-format off
+//   we turn off clang-format, so that the line numbers don't get changed by reformatting.
```

Comment:
That's strange. How is that different to any modifications made to the file? Or is it supposed that this file will *never* be modified?

Reply:
If you are working on the file, you might end up moving lines around. It's easy to see which line the `code_location::current()` call is on and update the calls in the file. But then, later, if clang-format is run, it may change the format and you have to update it _again_.  

It's very frustrating to get the test working, `git add` `git clang-format` `git commit` and then have the test fail because clang-format "helped" you.

Reply:
i incorporated your suggestion.